addons/hr/wizard/hr_departure_wizard.py

```diff
@@ -17,14 +16,32 @@ def _get_default_departure_date(self):
 
         return departure_date or fields.Date.today()
 
-    departure_reason_id = fields.Many2one("hr.departure.reason", default=lambda self: self.env['hr.departure.reason'].search([], limit=1), required=True)
+    def _get_departure_reason_domain(self):
```

Comment:
Define an ir.rule instead of applying a domain on the field, like the following:

```xml
    <record id="ir_rule_hr_payroll_structure_multi_company" model="ir.rule">
        <field name="name">HR Payroll Structure: Multi Company</field>
        <field name="model_id" ref="model_hr_payroll_structure"/>
        <field name="domain_force">['|', ('country_id', '=', False), ('country_id', 'in', user.env.companies.mapped('country_id').ids)]</field>
    </record>
```
That way, the domain will be automatically applied to all the foreign keys to this model (aka Many2one).